changelog/3191.feature.rst

```diff
@@ -0,0 +1,16 @@
+A warning is now issued when assertions are made directly against ``None``.
+
+This is a common source of confusion among new users, which write::
+
+    assert mocked_object.assert_called_with(3, 4, 5, key='value')
+
+When they should write::
+
+    mocked_object.assert_called_with(3, 4, 5, key='value')
+
+Because the ``assert_called_with`` method of mock objects already executes an assertion.
+
+This warning will not be issued when ``None`` is explicitly checked
+    assert none_returning_fun() is None
+
+will not issue the warning
```

Comment:
That's because you wrote it - https://github.com/pytest-dev/pytest/pull/4146/commits/26d27df6fce16904ed9fb27e27e16dc9620d9569 
I think I added that line to trigger the [pre-commit auto format](https://github.com/pytest-dev/pytest/pull/4146#issuecomment-434440703) not the best plan in hindsight

Reply:
I this section, hopefully it's clearer. If you still think it's out of place I'll delete it

Reply:
Ahh forgot about that, well at least I'm consistent then. It would be amusing if I looked at it and request a lot of changes. 😆

Reply:
Looks good, thanks!